apps/mobile-wallet/src/components/ConsolidationModal.tsx

```diff
@@ -25,43 +25,55 @@ import AppText from '~/components/AppText'
 import Button from '~/components/buttons/Button'
 import ButtonsRow from '~/components/buttons/ButtonsRow'
 import { ModalScreenTitle, ScreenSection } from '~/components/layout/Screen'
-import { ModalContent, ModalContentProps } from '~/features/modals/ModalContent'
+import BottomModal from '~/features/modals/BottomModal'
+import { closeModal } from '~/features/modals/modalActions'
+import { ModalContent } from '~/features/modals/ModalContent'
+import withModal from '~/features/modals/withModal'
+import { useAppDispatch } from '~/hooks/redux'
 
-interface ConsolidationModalProps extends ModalContentProps {
+interface ConsolidationModalProps {
   onConsolidate: () => void
   fees: bigint
 }
 
-const ConsolidationModal = ({ onConsolidate, fees, ...props }: ConsolidationModalProps) => {
+const ConsolidationModal = withModal<ConsolidationModalProps>(({ id, onConsolidate, fees }) => {
   const { t } = useTranslation()
+  const dispatch = useAppDispatch()
+
+  const handleConsolidate = () => {
+    onConsolidate()
+    dispatch(closeModal({ id }))
+  }
 
   return (
-    <ModalContent verticalGap {...props}>
-      <ScreenSection>
-        <ModalScreenTitle>{t('Consolidation required')}</ModalScreenTitle>
-      </ScreenSection>
-      <ScreenSection>
-        <View>
-          <AppText>
-            {t(
-              'It appears that the address you use to send funds from has too many UTXOs! Would you like to consolidate them? This will cost as small fee.'
-            )}
-          </AppText>
-          <Fee>
-            <AppText>{t('Fee')}:</AppText>
-            <Amount value={fees} fullPrecision fadeDecimals bold />
-          </Fee>
-        </View>
-      </ScreenSection>
-      <ScreenSection centered>
-        <ButtonsRow>
-          <Button title={t('Cancel')} onPress={props.onClose} flex variant="accent" short />
-          <Button title={t('Consolidate')} onPress={onConsolidate} variant="highlight" flex short />
-        </ButtonsRow>
-      </ScreenSection>
-    </ModalContent>
+    <BottomModal modalId={id}>
```

Comment:
Comment about the whole PR: every time I've wrapped a modal in `BottomModal`, the whole component is marked as changed, but it's not. 

What you should take a look at it how I extracted the callbacks, like `handleConsolidate` above. The main difference is that the modals are now responsible for closing themselves once the callbacks are executed. Before, it was the parents which were handling this.